modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C

```diff
@@ -131,6 +134,36 @@ AbaqusUMATStress::initialSetup()
                "': Incremental strain quantities are not available. You likely are using a total "
                "strain formulation. Specify `incremental = true` in the tensor mechanics action, "
                "or use ComputeIncrementalSmallStrain in your input file.");
+
+  // Let's automatically detect uos and identify the one we are interested in.
+  // If there is more than one, we assume something is off and error out.
+  if (!isParamSetByUser("step_user_object"))
+  {
+    std::vector<const UserObject *> uos;
+    _fe_problem.theWarehouse().query().condition<AttribSystem>("UserObject").queryInto(uos);
```

Comment:
Do a `queryIntoUnsorted` here to avoid an unnecessary dependency resolution. You might think it's not an "important" optimization, but your call will enforce a stricter detection of cyclical dependencies than what MOOSE currently does (only within execute on groups). I observed exactly that in a current PR where an external app suddenly failed with a cyclic dependency error.